app/controllers/custom_wizard/admin/wizard.rb

```diff
@@ -26,6 +26,8 @@ def show
 
   def remove
     if CustomWizard::Template.remove(@wizard.id)
+      related_custom_fields = CategoryCustomField.where(name: 'create_topic_wizard', value: @wizard.name)
+      related_custom_fields.destroy_all
```

Comment:
@merefield Please move this into the CustomWizard::Template.remove method itself, inside the transaction block in that method.

Reply:
Ah!  Good catch, apologies. Always good practice to minimise logic in controller (and take advantage of transaction).